src/hotspot/share/runtime/thread.hpp

```diff
@@ -134,6 +135,29 @@ class Thread: public ThreadShadow {
     return offset;
   }
 
+ protected:
+  // Poll data is used in generated code for safepoint polls.
+  SafepointMechanism::ThreadData _poll_data;
```

Comment:
Should this be private and you can add `friend class JavaThread` to this class. Also, the accessor below should be moved to `JavaThread`, too. Comments explaining that it is put here only for offset optimization would be great.

Reply:
I tried to avoid this kind of coupling between `Thread` and `JavaThread`, but I don't mind go that way as well. This is would also avoid the awkwardness of having `JavaThread::polling_data_*()` accessors all around our code calling essentially the superclass accessor otherwise. See new commit. I am re-running tests now.

Reply:
My thought is that it would make everything else see `poll_data` being a member of `JavaThread`, which seems more appropriate, while the fact that it is actually a member of `Thread` is merely an implementation detail.

Reply:
Right. I modeled this after nmethod barriers, which also (?) technically belong to Java threads only. But there is no need to spend more time on this :)